.github/workflows/rust.yml

```diff
@@ -2,9 +2,13 @@ name: Rust CI
 
 on:
   push:
-    branches: [ main ]
+    paths:
+      - 'rust/**'
+      - 'test/**'
   pull_request:
-    branches: [ main ]
```

Comment:
Why are you deleting main here? Usually prs should run on a PR against main, or on merge to main, but not on all branches